src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -636,14 +572,83 @@ EventEngine::ConnectionHandle PosixEventEngine::Connect(
          status = socket.status()]() mutable { on_connect(status); });
     return EventEngine::ConnectionHandle::kInvalid;
   }
-  return ConnectInternal((*socket).sock, std::move(on_connect),
-                         (*socket).mapped_target_addr,
-                         std::move(memory_allocator), options, timeout);
+  return CreateEndpointFromUnconnectedFd(
+      (*socket).sock.Fd(), std::move(on_connect), (*socket).mapped_target_addr,
+      args, std::move(memory_allocator), timeout);
```

Comment:
Every connection attempt will now be doing the EndpointConfig -> PosixTcpOptions conversion twice. Since you want the new public API to take `EndpointConfig`, I'd suggest adding a thin layer: have `CreateEndpointFromUnconnectedFd` just convert the config to PosixTcpOptions, and proxy those args to `CreateEndpointFromUnconnectedFdInternal`.

```
EventEngine::ConnectionHandle PosixEventEngine::CreateEndpointFromUnconnectedFdInternal(
    int fd, EventEngine::OnConnectCallback on_connect,
    const EventEngine::ResolvedAddress& addr, const PosixTcpOptions& options,
    MemoryAllocator memory_allocator, EventEngine::Duration timeout) {
```